akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/HttpChallenge.scala

```diff
@@ -22,6 +23,16 @@ final case class HttpChallenge(scheme: String, realm: String,
   def getParams: util.Map[String, String] = params.asJava
 }
 
+// FIXME: AbstractFunction3 required for bin compat. remove in Akka 3.0 and change realm in case class to option #20786
```

Comment:
Yer, for the benefit of others reading this. Since this approach replaces the generated companion for HttpChallenge its necessary to extend AbstractFunction3 as the scala compiler does when it generates the companion